package.json

```diff
@@ -114,10 +114,9 @@
     "@rtsao/plugin-proposal-class-properties": "7.0.1-patch.1",
     "@swc/core": "1.3.38",
     "@swc/helpers": "0.4.14",
-    "@testing-library/dom": "8.20.0",
+    "@testing-library/dom": "9.0.1",
```

Comment:
RTL is built on top `testing-library/dom`, so should include all its functions. I quickly checked our codebase and it seems like we use `testing-library/dom` mostly for the `within` function, which is also available from `testing-library/react`. Should we try to remove this package?

Reply:
ok so i've changed the instances of the code that are using `within`/`fireEvent` from `@testing-library/dom` directly to instead use the methods from `@testing-library/react`, however i don't think we can remove the package itself as `@testing-library/user-event` declares a peer dependency on it.

Reply:
I think `@tessting-library/react` should install it? Also found [an issue](https://github.com/testing-library/user-event/issues/1101) where it says that `@testing-library/dom` shouldn't be listed as dependency unless used directly.

Reply:
it does, but that will still leave us with warnings when yarn installing:
```
➤ YN0002: │ grafana@workspace:. doesn't provide @testing-library/dom (p15dc6), requested by @testing-library/user-event
```

there's another issue [here](https://github.com/testing-library/user-event/issues/551#issuecomment-770337264) saying that installing it directly is the correct approach when using `@testing-library/user-event`.